### PR TITLE
Make error parsing more robust

### PR DESCRIPTION
**Pull request by kyleknap**
For the rest-xml error parser, the body of the response had headers of a
200 response even though the http status code was 500. This blew up
of the error parser because it was invalid xml. For these situations, we should
at least respect the error code.

cc @jamesls @mtdowling @rayluo @JordonPhillips

### CODE REVIEW

**Comment by JordonPhillips** [comment]
LGTM
:ship:

**Comment by rayluo** [comment]
:ship:

File: botocore/parsers.py [hunk 1]
```diff
@@ -674,9 +674,14 @@ def _do_error_parse(self, response, shape):
...
-            return self._parse_error_from_body(response)
-        else:
-            return self._parse_error_from_http_status(response)
+            # If the body ends up being invalid xml, the xml parser should not
+            # blow up. It should at least try to pull information about the
+            # the error response from other sources like the HTTP status code.
+            try:
+                return self._parse_error_from_body(response)
+            except ResponseParserError:
+                pass
```
Comment (jamesls) [inline botocore/parsers.py]: We should probably debug log this when it happens (with the traceback included).

**Comment by kyleknap** [comment]
@jamesls 
I agree with adding a debug log for the caught exception. Updated the PR. Ready to be looked at again.

**Comment by kyleknap** [comment]
@jamesls Updated debug log statement.

**Comment by mtdowling** [comment]
:shipit: this is similar to what some of the other SDKs are doing when parsing errors.

**Comment by jamesls** [comment]
:shipit:

**Comment by kyleknap** [comment]
I updated the comment in the test as well to reflect the unknown cause of the issue and the fact that it may be fixed in the future.

**Pull request merged**